dot/network/errors.go

```diff
@@ -0,0 +1,17 @@
+// Copyright 2021 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package network
+
+import (
+	"errors"
+)
+
+var (
+	errCannotValidateHandshake     = errors.New("failed to validate handshake")
+	errInvalidNotificationsMessage = errors.New("message is not NotificationsMessage")
```

Comment:
nit maybe change that to `errMessageTypeNotValid = errors.New("message type is not valid")`

and use it as `fmt.Errorf("%w: expected %T but got %T", errMessageTypeNotValid, (NotificationsMessage)(nil), message)`? That would be cool since it would adapt to name changes to the types. But that might be me getting excited for no reason haha!